core/src/main/scala/filodb.core/memstore/TimeSeriesShard.scala

```diff
@@ -429,6 +440,19 @@ class TimeSeriesShard(val ref: DatasetRef,
             case e: Exception                   => logger.error(s"Unexpected ingestion err", e); disableAddPartitions()
           }
         } else {
+          val oldTimestamp = groupTimestamps(group)
+          val newTimestamp = Math.max(oldTimestamp, ingestionTime) // monotonic clock
+          groupTimestamps(group) = newTimestamp
+          if (newTimestamp > oldTimestamp && oldTimestamp != Long.MinValue
+              && (oldTimestamp / flushBoundaryMillis) != (newTimestamp / flushBoundaryMillis)) {
+            flushSched match {
+              case None => // none provided, so don't flush
+              case Some(sched) => {
+                // Flush out the group first such that the record is for a new hour.
+                createFlushTask(prepareFlushGroup(group)).executeOn(sched)
```

Comment:
This logic does things on a different thread, and should not be here - it pollutes the IngestConsumer.

Reply:
It looks like the original task was scheduled on the same thread. I'll check again.